### DISCUSSION TITLE
R4S挂载USB机械硬盘负载稍高就挂掉

### DISCUSSION THREAD

**Discussion by wolvth** [istoreos/istoreos#671] [Q&A]
5v3a电源，一个2.5寸机械盘。aria2，局域网其他电脑迅雷往挂载的硬盘下载，时间长一些，或者下载速度大于40M就很容易把usb整挂。
320这个版本之前有开启沙盒，那时候如果usb挂掉后过一会儿还会自动恢复。
更新后发现沙盒没了，就没用沙盒模式。usb挂了后就恢复不了。
首页磁盘信息里的TF卡信息，挂载的硬盘信息都没有，挂载点和磁盘管理也打不开。
有时候会连web管理页面也进不去，ssh进去reboot没用。只能断电重启。
不知道是否因为电源不足导致，最近买了5v4a的电源，收货后再来反馈，看下有没其他r4s的朋友有同样的症状。

**Comment by jjm2473** [comment]
奇怪，最近不少人反馈沙箱没了，但是最近没改沙箱啊

**Reply by jjm2473** [reply]
你看看这个文件在不在 /usr/sbin/sandbox

**Reply by wolvth** [reply]
> 你看看这个文件在不在 /usr/sbin/sandbox

我说的沙箱没了是说自动退出了沙箱模式。功能还是在的，就是所有配置都没了。你指的文件是root下的话，这个文件还在。
之前的沙箱所在盘的配置文件夹我看退出了，就全删掉纯当个下载盘了。

**Reply by jjm2473** [reply]
哦，只是因为u盘挂了，所以退出了沙箱

**Reply by wolvth** [reply]
> 哦，只是因为u盘挂了，所以退出了沙箱

你这么说也不对，我在升级之前usb接口偶尔挂了（不确定真挂还是卡住）但是会自动连上，这时候沙盒模式也没自动退出，就算重启后还是会在沙盒模式里。只是因为升级后沙盒模式自动退出了

**Reply by jjm2473** [reply]
OK，我把上次改的usb补丁会滚掉，应该就没问题了。

**Reply by jjm2473** [reply]
有空测试下这个固件，看USB会不会稳定一些 https://we.tl/t-ilEfSrTiFL

**Reply by jjm2473** [reply]
之前你是只接了U盘，最近才接硬盘吗？

**Reply by wolvth** [reply]
> 之前你是只接了U盘，最近才接硬盘吗？

我没接过u盘，是usb移动硬盘2.5寸，用硬盘盒接的。现在电源是5v4a。
我问题里描述的硬盘都是指的usb外接的移动机械硬盘
-----------
你刚发我的固件，我测试了些时间，还是不稳定。
局域网电脑往硬盘smb传一个70G的文件，来回反复传了9次吧，出现过一次挂壁，
aria2 下载阿凡达2，30G的磁链，必定挂壁。
局域网电脑迅雷下载阿凡达2磁链，文件保存位置为外接硬盘也会挂壁
插拔一次usb外接硬盘就全都恢复，有时候卡住了，我在web页面操作多了会把web搞挂
ssh reboot会失效或者重启不起来，拔掉硬盘就好了，或者整个r4s断电一次才会恢复。

ssh进去df -h看的到所有硬盘信息，fdisk -l 只看的到TF卡，本该出来外接硬盘的信息那里会卡住
我把系统日志，内核日志复制下来了，你可以看下。
https://we.tl/t-huiQcNwU5A

**Reply by jjm2473** [reply]
测试固件和之前的0320固件都不稳定吗？

**Reply by wolvth** [reply]
> 测试固件和之前的0320固件都不稳定吗？

 是的

**Reply by jjm2473** [reply]
那应该没办法了，测试固件的usb配置已经跟openwrt一致了，0320则是修改了usb

**Reply by jjm2473** [reply]
后来怎么样，换电源有用吗

**Reply by wolvth** [reply]
换电源也没用，后续就懒的试了，因为有nas，这个挂载硬盘只是轻量使用不是刚需。有几个月没重启更新固件了，所以现在新固件会怎么样还不知道。

**Reply by nzhijun** [reply]
N1盒子出现了同样的状况，挂载的还是闲置固态，发现只要cpu占用百分百就很快挂掉。感觉是个系统调度问题。。。

**Reply by Kaouning** [reply]
我系统用了omv、友善的openwrt、istore、armbian、casaos均会掉盘，读写量上去了无解，电源换了2个，一个超稳定5v3a电压波动5%以内，一个不会低于5v的5.3v4a，事实证明5.3v那个相对reboot命令来说要稳定得多，但读写负载上去了均会掉盘（但之前下载一个200g文件写入飙升80M/s都稳定得一批就很奇怪），我还是装的4t的ssd，功率2w左右，固态u盘从来不掉，功率1-1.5w，换了2个硬盘盒子，均无效，很怀疑是电源波动导致的问题（处于某一个功率区间就很稳定），但没办法测试，准备买一个带电压稳定芯片的usb线试试，但估计是无效果的

**Reply by kxchen** [reply]
最新固件，N5105挂载移动硬盘稍微上点速度，必掉盘。无任何读写操作也几个小时掉一次。
